### DISCUSSION TITLE
Problems in understanding GTBS - TBS relation in subsys/bluetooth/audio/tbs.c

### DISCUSSION THREAD

**Discussion by lukas-fwdev** [zephyrproject-rtos/zephyr#71883] [Q&A] [answered]
Hi community,

I am currently trying to understand how the Generic TBS and TBS instances are handled in the implementation, especially how the init is done in 'subsys/bluetooth/audio/tbs.c' bt_tbs_init() function. There I can see that bt_gatt_service_register() is only done on the tbs instances but not for gtbs, right? I would like to know if this is by intention or a bug? 
I can see that there had been a lot of changes done by @MariuszSkamra and @fabiobaltieri some months ago, maybe you can help?

Any help understanding this deeper is very much appreciated.

Thanks in advance!
lukas

**Comment by fabiobaltieri** [comment]
cc @Thalley

**Comment by Thalley** [comment, marked as answer]
GTBS is statically registered via 
```c
BT_GATT_SERVICE_DEFINE(gtbs_svc, BT_TBS_SERVICE_DEFINE(BT_UUID_GTBS, &gtbs_inst.inst));
```
when `CONFIG_BT_GTBS=y`. 

The other TBS uses `bt_gatt_service_register` because at the time of the implementation, we did not have a way of statically registering multiple services (we do now, but haven't been implemented). 

It's a moot point though, because we want to move towards dynamically registering them: https://github.com/zephyrproject-rtos/zephyr/issues/41738

**Reply by lukas-fwdev** [reply]
Thanks for the fast response! This was the information I was looking for.

The proposed enhancement would indeed make things a lot easier and dealing with more TBS instances much more flexible.
